esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2218,32 +2218,39 @@ void CFileSprayEx::getDropZoneInfoByIP(const char* ip, const char* destFileIn, S
     if (!env)
         return;
 
-    Owned<IConstMachineInfo> machine = env->getMachineByAddress(ip);
-    if (!machine)
-    {
-        IpAddress ipAddr;
-        ipAddr.ipset(ip);
-        if (!ipAddr.isLocal())
-            return;
-        machine.setown(env->getMachineForLocalHost());
-        if (!machine)
-            return;
-    }
-    SCMStringBuffer computer, directory, maskBuf;
-    machine->getName(computer);
-    if (!computer.length())
-        return;
-
-    Owned<IConstDropZoneInfo> dropZone = env->getDropZoneByComputer(computer.str());
-    if (!dropZone)
-        return;
-    dropZone->getDirectory(directory);
+    SCMStringBuffer directory;
+    Owned<IConstDropZoneInfo> dropZone;
 
     StringBuffer destFile;
     if (isAbsolutePath(destFileIn))
+    {
         destFile.set(destFileIn);
+        dropZone.setown(env->getDropZoneByAddressPath(ip, destFile.str()));
+        if (!dropZone)
+            return;
+        dropZone->getDirectory(directory);
+    }
     else
     {
+        SCMStringBuffer computer;
+        Owned<IConstMachineInfo> machine = env->getMachineByAddress(ip);
+        if (!machine)
+        {
+            IpAddress ipAddr;
+            ipAddr.ipset(ip);
+            if (!ipAddr.isLocal())
+                return;
+            machine.setown(env->getMachineForLocalHost());
+            if (!machine)
+                return;
+        }
+        machine->getName(computer);
+        if (!computer.length())
+            return;
+        dropZone.setown(env->getDropZoneByComputer(computer.str()));
```

Comment:
This code uses getDropZoneByAddressPath(ip, path) to locate the DZ if an abs path is provided.
If an abs path is not provided then it does what it used to do - finding the first DZ that matches the machine.
Is getDropZoneByAddressPath() ok to call ?